### PR TITLE
Revised addition operator reference

### PR DESCRIPTION
**Pull request by pkulikov**
Revised addition operators (`+` and `+=`) docs:
- move snippets to the samples repo: dotnet/samples#404
- split the `+` topic in sections
- add references to the C# language spec

### CODE REVIEW

**Comment by pkulikov** [comment]
As snippets PR is merged, let's close/reopen.

**Comment by rpetrusha** [comment]
Build warnings show that merge of examples wasn't recognized. Closing and reopening to begin new build.

File: docs/csharp/language-reference/operators/addition-assignment-operator.md [hunk 1]
```diff
@@ -1,40 +1,46 @@
...
-  
- The `+=` operator is also used to specify a method that will be called in response to an event; such methods are called event handlers. The use of the `+=` operator in this context is referred to as *subscribing to an event*. For more information, see [How to: Subscribe to and Unsubscribe from Events](../../../csharp/programming-guide/events/how-to-subscribe-to-and-unsubscribe-from-events.md) and [Delegates](../../../csharp/programming-guide/delegates/index.md).  
-  
-## Example  
- [!code-csharp[csRefOperators#35](../../../csharp/language-reference/operators/codesnippet/CSharp/addition-assignment-operator_1.cs)]  
+For numeric types, the [addition operator](addition-operator.md) `+` computes the sum of its operands. If one or both operands of type [string](../keywords/string.md), it concatenates the string representations of its operands. For delegate types, the `+` operator returns a new delegate instance that is combination of its operands.
```
Comment (rpetrusha) [inline docs/csharp/language-reference/operators/addition-assignment-operator.md]: of type --> is of type

File: docs/csharp/language-reference/operators/addition-operator.md [hunk 2]
```diff
@@ -10,26 +10,49 @@ helpviewer_keywords:
...
-## See Also
-
-- [C# Reference](../../../csharp/language-reference/index.md)  
-- [C# Programming Guide](../../../csharp/programming-guide/index.md)  
-- [C# Operators](../../../csharp/language-reference/operators/index.md)  
-- [operator (C# Reference)](../../../csharp/language-reference/keywords/operator.md)
+
+The `+` operator is supported in two forms: a unary plus operator or a binary addition operator.
+
+User-defined types can [overload](../keywords/operator.md) either of unary or binary `+` operators. When a binary `+` operator is overloaded, the [addition assignment operator](addition-assignment-operator.md) `+=` is also implicitly overloaded.
```
Comment (rpetrusha) [inline docs/csharp/language-reference/operators/addition-operator.md]: either of unary or binary --> the unary and binary

File: docs/csharp/language-reference/operators/addition-operator.md [hunk 3]
```diff
@@ -10,26 +10,49 @@ helpviewer_keywords:
...
+
+## Unary plus operator
+
+Unary `+` operator returns the value of its operand. It's supported by all numeric types.
```
Comment (rpetrusha) [inline docs/csharp/language-reference/operators/addition-operator.md]: Unary --> The unary

File: docs/csharp/language-reference/operators/remainder-operator.md [hunk 4]
```diff
@@ -27,7 +29,7 @@ For the [float](../keywords/float.md) and [double](../keywords/double.md) operan
 - the sign of `z`, if non-zero, is the same as the sign of `x`;
 - the absolute value of `z` is the value produced by `|x| - n * |y|` where `n` is the largest possible integer that is less than or equal to `|x| / |y|` and `|x|` and `|y|` are the absolute values of `x` and `y`, respectively.
 
-For information about behavior of the `%` operator in case of non-finite operands, see the [Remainder operator](~/_csharplang/spec/expressions.md#remainder-operator) section of the [C# language specification](/dotnet/csharp/language-reference/language-specification/index).
+For information about behavior of the `%` operator in case of non-finite operands, see the [Remainder operator](~/_csharplang/spec/expressions.md#remainder-operator) section of the [C# language specification](../language-specification/index.md).
```
Comment (rpetrusha) [inline docs/csharp/language-reference/operators/remainder-operator.md]: unrelated to your PR: 
- behavior --> the behavior
- in case of --> with

**Review by rpetrusha — CHANGES_REQUESTED** [review]
This looks good, @pkulikov. Thanks also for moving the code to the samples repo. I've left a couple of suggested changes.

**Comment by pkulikov** [comment]
@rpetrusha thanks for review; I've addressed your feedback.

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
Thanks, @pkulikov, for the additional changes. I'll merge your PR now.

**Pull request merged**